src/app/timeseries-menu.component.ts

```diff
@@ -98,6 +106,55 @@ export class TimeseriesMenuComponent {
     }
   }
 
+  getToolTipData(jsonString) {
+    this.toolTipData = [];
+    const jsonArr = jsonString.split("{"); // counter,, jsonArr.length = n + 1 where n is number of layers selected
+    for (let i = 1; i < jsonArr.length; i++) {
+      const v = jsonArr[i].split('value": ');
+      const d = jsonArr[i].split('name": ');
+      const series = jsonArr[i].split('series": ');
+      const l = series[1].split('"'); // layer
+      v[1] = v[1].substr(0, 4); // value
+      d[1] = d[1].substr(1, 7); // date
+      this.toolTipData.push({
+        date: d[1],
+        value: v[1],
+        layer: l[1]
+      });
+    }
+    return this.toolTipData;
+  }
+
+  chartToolTipDate(jsonString) {
```

Comment:
I'm not sure why you have a json string here. But you shouldn't be parsing the json as a string. You instead should just convert it to a class/object and get the elements out of it normally:
`JSON.parse(jsonString)`